sync-core/src/test/java/com/cloudant/sync/query/QueryTextSearchTest.java

```diff
@@ -0,0 +1,233 @@
+
+        MutableDocumentRevision rev = new MutableDocumentRevision();
+        rev.docId = "mike12";
+        Map<String, Object> bodyMap = new HashMap<String, Object>();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 12);
+        bodyMap.put("pet", "cat");
+        bodyMap.put("comment", "He lives in Bristol, UK and his best friend is Fred.");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        ds.createDocumentFromRevision(rev);
+
+        rev.docId = "mike34";
+        bodyMap.clear();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 34);
+        bodyMap.put("pet", "dog");
+        bodyMap.put("comment", "He lives in a van down by the river in Bristol.");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        ds.createDocumentFromRevision(rev);
+
+        rev.docId = "mike72";
+        bodyMap.clear();
+        bodyMap.put("name", "mike");
+        bodyMap.put("age", 72);
+        bodyMap.put("pet", "cat");
+        bodyMap.put("comment", "He's retired and enjoys spending time with his cat Remus.");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        ds.createDocumentFromRevision(rev);
+
+        rev.docId = "fred34";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 34);
+        bodyMap.put("pet", "cat");
+        bodyMap.put("comment",
+                    "He lives next door to Mike and his cat Romulus is brother to Remus.");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        ds.createDocumentFromRevision(rev);
+
+        rev.docId = "fred12";
+        bodyMap.clear();
+        bodyMap.put("name", "fred");
+        bodyMap.put("age", 12);
+        bodyMap.put("comment", "He lives in Bristol, UK and his best friend is Mike.");
+        rev.body = DocumentBodyFactory.create(bodyMap);
+        ds.createDocumentFromRevision(rev);
+    }
+
+    @Test
+    public void canMakeAQueryConsistingOfASingleTextSearch() {
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("comment"), "basic_text", "text"),
+                                    is("basic_text"));
+
+        // query - { "$text" : { "$search" : "best friend" } }
+        Map<String, Object> search = new HashMap<String, Object>();
+        search.put("$search", "lives in Bristol");
+        Map<String, Object> query = new HashMap<String, Object>();
+        query.put("$text", search);
+        QueryResult queryResult = im.find(query);
+        assertThat(queryResult.documentIds(), containsInAnyOrder("mike12", "mike34", "fred12"));
+    }
+
+    @Test
+    public void canMakeAQueryTextSearchUsingDoubleQuotes() {
+        assertThat(im.ensureIndexed(Arrays.<Object>asList("comment"), "basic_text", "text"),
+                is("basic_text"));
+
+        // query - { "$text" : { "$search" : "\"lives in Bristol\"" } }
+        Map<String, Object> search = new HashMap<String, Object>();
+        search.put("$search", "\"lives in Bristol\"");
```

Comment:
Is this a phrase search? If so perhaps make the test name more focused on the type of search, `canMakePhraseSearch`?